### ISSUE TITLE
Separate data provider dependencies from Elmah Core

### ISSUE THREAD

**Issue by GoogleCodeExporter** [Benknightdark/elmah#292]
```
What new or enhanced feature are you proposing?

Please change the structure of the Elmah project so that the core DLL does not 
have a dependency on specific data providers (MySql.Data, Npgsql, 
System.Data.SqlServerCe, System.Data.SQLite).

What goal would this enhancement help you achieve?

Removing tightly-coupled dependencies is considered good practice. Also, every 
time we run our unit tests we're getting "Test run warnings" about these 
assemblies not being found, even though we don't use them.
```

Original issue reported on code.google.com by `Striplin...@gmail.com` on 18 Apr 2012 at 11:34

**Comment by GoogleCodeExporter** [comment]
```
As a suggestion, I'd propose you use Nuget?

In there, there is an Elmah.core, and then you can add other packages like 
elmah.xml.
For example, this package: http://nuget.org/packages/elmah.xml contains a 
dependency to elmah, which depends on the 
http://nuget.org/packages/elmah.corelibrary. 

The core library is built to not have any dependencies. The extra packages then 
add providers, like SQL and XML.
```

Original comment by `mich...@mbwarez.dk` on 25 May 2012 at 11:01

**Comment by GoogleCodeExporter** [comment]
```
I just converted our projects to use Nuget, and this *does not* resolve the 
issue.

You say that the core library is built to not have any dependencies, but in 
fact ILSpy shows that Elmah.dll depends on all the libraries mentioned in the 
initial issue report. So our test runs still produce warnings.
```

Original comment by `Striplin...@gmail.com` on 4 Jun 2012 at 4:12

Attachments:
- [ElmahDependencies.PNG](https://storage.googleapis.com/google-code-attachments/elmah/issue-292/comment-2/ElmahDependencies.PNG)
- [testrunwarnings.PNG](https://storage.googleapis.com/google-code-attachments/elmah/issue-292/comment-2/testrunwarnings.PNG)

**Comment by GoogleCodeExporter** [comment]
```
Confirmed. I have elmah.corelibrary.1.2.2, and ILSpy on my dll shows the same 
dependencies as the above screenshot.

The warning doesn't seem to cause any problems, but is obviously not a good 
thing to have hanging around.
```

Original comment by `visse...@gmail.com` on 30 Aug 2013 at 5:44

**Comment by GoogleCodeExporter** [comment]
```
This is currently in experimentation at the following clone:
http://code.google.com/r/azizatif-elmah-split/
```

Original comment by `azizatif` on 30 Aug 2013 at 5:59
- Changed state: **Started**
- Added labels: **Milestone-Release2.0**

**Comment by GoogleCodeExporter** [comment]
```
This issue was closed by revision 938bc80cf3f7.
```

Original comment by `azizatif` on 18 Dec 2013 at 5:37
- Changed state: **Fixed**
